src/app/GitExtUtils/GitUI/ComboBoxExtensions.cs

```diff
@@ -21,37 +23,72 @@ public static void AdjustWidthToFitContent(this ComboBox comboBox)
 
         public static void ResizeDropDownWidth(this ComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
             int calculatedWidth = GetPreferredDropDownWidth(comboBox);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
         public static void ResizeDropDownWidth(this ToolStripComboBox comboBox, int minWidth, int maxWidth)
         {
-            if (comboBox is null)
-            {
-                throw new ArgumentNullException(nameof(comboBox));
-            }
+            ArgumentNullException.ThrowIfNull(comboBox);
 
-            int calculatedWidth = GetPreferredDropDownWidth(comboBox.Control);
+            int calculatedWidth = GetPreferredDropDownWidth((ComboBox)comboBox.Control);
             comboBox.DropDownWidth = Math.Min(Math.Max(calculatedWidth, minWidth), maxWidth);
         }
 
-        private static int GetPreferredDropDownWidth(dynamic comboBox)
+        private static int GetPreferredDropDownWidth(ComboBox comboBox)
         {
             int calculatedWidth = 0;
-            using dynamic graphics = comboBox.CreateGraphics();
-            foreach (object obj in comboBox.Items)
+            using Graphics graphics = comboBox.CreateGraphics();
+
+            string displayMemberName = comboBox.DisplayMember;
+            Font font = comboBox.Font;
+
+            foreach (object item in comboBox.Items)
             {
-                dynamic area = graphics.MeasureString(obj.ToString(), comboBox.Font);
+                SizeF area = graphics.MeasureString(GetDisplayValue(displayMemberName, item), font);
                 calculatedWidth = Math.Max((int)area.Width, calculatedWidth);
             }
 
-            return calculatedWidth;
+            return calculatedWidth + CalculateVerticalScrollBarWidth(comboBox);
+        }
+
+        private static string GetDisplayValue(string displayMemberName, object item)
+        {
+            if (displayMemberName.Length == 0)
+            {
+                return item.ToString();
+            }
+
+            System.Reflection.PropertyInfo displayMemberProperty = item
+                .GetType()
+                .GetProperty(displayMemberName);
+
+            if (displayMemberProperty is null)
+            {
+                return item.ToString();
+            }
+
+            return displayMemberProperty
+                .GetValue(item)
+                ?.ToString();
```

Comment:
This would change semantics of how `ComboBox` displays items. `ComboBox` accepts (doesn't throw) if a value from a property defined by `DisplayMember` is null. An empty item is displayed in such case and this is different from `item.ToString()` value.

Reply:
The first `?.` does the trick. Take a second look, please.

Reply:
I hacked up a demo of why this is semantically not equivalent to initial implementation:

I've configured the ComboBox to take list item values from `ForDisplay` field, meaning `ToString` should never be used on it... alas..

![image](https://github.com/user-attachments/assets/05ab284d-3738-4315-8892-d5a4f0422176)

![image](https://github.com/user-attachments/assets/9680b28a-a8aa-4240-922a-ac60a35d633a)

Reply:
In the moment of my review, this function was supposed to return a non-null value.
As this has changed, you are right.
Following this approach, `item.ToString()` must not be called, too, in case of `displayMemberProperty is null`, i.e.
```suggestion
            return displayMemberProperty
                ?.GetValue(item)
                ?.ToString();
```

Reply:
p.s. this is how the list looks with these values. As you can see the one with the `null` is shown as an empty item while the suggested code would measure the length of the string `WRONG VALUE` which would not match what is displayed.

![image](https://github.com/user-attachments/assets/04b84ab3-b8bf-4d3d-a2ab-d2508d44e754)

Reply:
Michael, why are you trying to break my code? 😅

Here is an example where there are two types of items and only some of those have properties configured by `DisplayMember`. When an item doesn't have a property `ComboBox` falls back to `ToString` while the suggested code measures a length of `null`

![image](https://github.com/user-attachments/assets/6ce35b38-4596-43c0-b7f5-8ec0049f0c90)

![image](https://github.com/user-attachments/assets/342f479f-8936-4cf3-9849-1d8dfa4219ad)

Here a value of `ToString` is displayed but the length of `null` was measured

![image](https://github.com/user-attachments/assets/29a7aa4b-9b13-421e-8cc7-9db017296a24)

Reply:
> Michael, why are you trying to break my code?

I just wanted to shorten the code. :)

A comment "When an item doesn't have a property ComboBox falls back to ToString - but not if the property contains null." will help future readers of the code.
(The detailed behavior of UI elements is often tricky.)

Reply:
Comment added 👍
yeah... I've found out this behavior through testing these edge cases.. I had a bug here as I didn't anticipate items of different types at first. Once I got bitten by it I've done some behavioral exploration :)

Reply:
It also means we need more tests to cover all these use cases 😉

Reply:
Added some tests :)